app/controllers/procedures_controller.rb

```diff
@@ -0,0 +1,36 @@
+class ProceduresController < ApplicationController
+  def show
+    @proposal = Proposal.find(params[:proposal_id])
```

Comment:
use before_action and create methods for setting proposal, experiment and procedure. Look at the experiment controller for examples.